src/libs/Network.js

```diff
@@ -32,7 +32,7 @@ let didLoadPersistedRequests;
 Onyx.connect({
     key: ONYXKEYS.NETWORK_REQUEST_QUEUE,
     callback: (persistedRequests) => {
-        if (didLoadPersistedRequests || !persistedRequests) {
+        if (didLoadPersistedRequests || !persistedRequests || persistedRequests.length === 0) {
```

Comment:
Is this correct? Based on the issue I just read it seems like `didLoadPersistedRequests` _should_ be preventing this callback from executing - but does not because we are calling `Onyx.set()` before updating that variable. LMK if I've got it wrong. Probably we should...

1. Move the `didLoadPersistedRequests` above the `Onyx.set()` in this function to solve the issue for now
2. Fix Onyx so that `keysChanged()` is async (what we have before) to prevent other potential issues where we have assumed an asynchronous callback

Reply:
Yes that solution also works, but is there a reason why we would want to continue running if have an empty array?

Reply:
I'm still happy to switch to just do the other change though. That way we'll have close to original behavior as possible.

Reply:
>is there a reason why we would want to continue running if have an empty array

No, I can't really think of one. But it's not the issue we are trying to fix here and doesn't seem to be causing any specific problem for us.